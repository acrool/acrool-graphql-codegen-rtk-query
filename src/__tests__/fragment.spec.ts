import {parse, buildSchema} from 'graphql';
import {plugin} from '../index';

const fragmentDoc = parse(/* GraphQL */ `
    fragment ResultIconSvg on IconSvg {
        id
        code
        viewBox
        content
        defs
    }
    query getIcon {
        iconSvg {
            ...ResultIconSvg
        }
    }
`);

describe('Fragment Plugin', () => {
    const schema = buildSchema(/* GraphQL */ `
        type Query {
            iconSvg: IconSvg!
        }
        type IconSvg {
            id: ID!
            code: String!
            viewBox: String
            content: String
            defs: String
        }
    `);
    const document = [{document: fragmentDoc}];

    it('should generate fragment and query correctly', async () => {
        const result = await plugin(schema, document, {
            importBaseApiFrom: '@/library/graphql/baseApi',
            importBaseApiAlternateName: 'baseApi',
            exportHooks: true,
            exportApi: true,
            exportDocument: true,
            exportApiName: 'api',
            exportDefaultApi: true,
        });
        expect(result).toMatchSnapshot();
    });
}); 