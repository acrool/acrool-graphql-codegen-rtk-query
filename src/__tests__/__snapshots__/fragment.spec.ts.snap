// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment Plugin should generate fragment and query correctly 1`] = `
{
  "content": "
export const GetIconDocument = \`
query getIcon {
  iconSvg {
    ...ResultIconSvg
  }
}
\${ResultIconSvg}\`;

const injectedRtkApi = baseApi.injectEndpoints({
  endpoints: (build) => ({
    getIcon: build.query<GetIconQuery, IUseFetcherArgs<GetIconQueryVariables> | void>({
      query: (args) => ({ document: GetIconDocument, args })
    }),
  }),
});

export default injectedRtkApi;
export const { useGetIconQuery, useLazyGetIconQuery } = injectedRtkApi;



",
  "prepend": [
    "import { baseApi } from '@/library/graphql/baseApi';",
    "import {gql, useSubscription, SubscriptionHookOptions} from '@apollo/client';",
  ],
}
`;
